compiler/src/dotty/tools/dotc/transform/CheckUnused.scala

```diff
@@ -569,7 +570,13 @@ object CheckUnused:
         defdef.symbol.is(Deferred) ||
         (rhs match {
           case _: tpd.Literal => true
-          case _ => rhs.tpe.isInstanceOf[ConstantType]
+          case _ => rhs.tpe match
```

Comment:
I wonder if looking at the `rhs.tpe` here is the "right" way to identify "trivial" bodies. A method can have a complex body, but if it returns a singleton unused parameters won't be reported:

```scala
object t {
  def m(x: Int, y: Int) = {
    println(s"non-trivial method body, only using $x")
    Nil
  }
}
```

I see that it's the same in Scala 2, which is a good approach. But I wonder if it leads to unwanted false negatvies.

Reply:
Is there a **formal** definition of trivial method? I couldn't find one except what I understood from the Scala 2 code?

Reply:
Shouldn't this also be a separate issue where a formal definition of trivial methods is given?

Reply:
I think it makes sense to create a ticket for it.